src/main/client/type.c

```diff
@@ -1416,7 +1416,8 @@ static int AerospikeClient_Type_Init(AerospikeClient *self, PyObject *args,
 
     self->as = aerospike_new(&config);
 
-    if (AerospikeClientConnect(self) == NULL) {
+    if (AerospikeClientConnect(self) == -1) {
+        aerospike_destroy(self->as);
```

Comment:
No, when the client constructor throws an exception because it failed to connect, it won't return a client object. The client constructor has to succeed in order to return a client object.

Reply:
```python
>>> import aerospike
>>> config = {"hosts": [("15.15.15.15", 3000)]}
>>> client = aerospike.client(config)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
exception.TimeoutError: (9, 'Failed to connect', 'src/main/aerospike/as_cluster.c', 415, False)
>>> key = ("test", "demo", 1)
>>> client.put(key, {"a": 1})
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
NameError: name 'client' is not defined
```